src/adapter/src/coord/timeline.rs

```diff
@@ -840,6 +840,12 @@ impl Coordinator {
         let mut id_bundle: CollectionIdBundle = self
             .index_oracle(compute_instance)
             .sufficient_collections(item_ids.iter());
+        // Queries may get auto-routed to the mz_introspection cluster, so we include indexes on
+        // that cluster.
+        let system_id_bundle = self
+            .index_oracle(*self.catalog().get_mz_introspections_cluster_id())
+            .sufficient_collections(item_ids.iter());
```

Comment:
Even if `item_ids` only contain items from the Postgres catalog schemas, `sufficient_collections` will still return indexes in the `mz_catalog` schema. So I just decided to always include all catalog schemas.